modules/utils/src/filesystem/file.cpp

```diff
@@ -0,0 +1,109 @@
+      !infile, fmt::format("Could not open file {} to read content", path.string()));
+
+  std::string text;
+  text.reserve(std::filesystem::file_size(path));
+  text.assign(std::istreambuf_iterator<char>(infile), std::istreambuf_iterator<char>());
+
+  return text;
+}
+
+auto readBinaryFile(const std::filesystem::path& path) -> std::vector<std::byte> {
+  std::ifstream infile{ path, std::ios::binary };
+  throwExceptionIf<InvalidDataException>(
+      !infile, fmt::format("Could not open file {} to read content", path.string()));
+
+  const auto file_size = std::filesystem::file_size(path);
+  std::vector<std::byte> buffer(file_size);
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+  infile.read(reinterpret_cast<char*>(buffer.data()), static_cast<std::streamsize>(file_size));
+
+  return buffer;
+}
+
+void writeStringToFile(const std::filesystem::path& path, std::string_view content) {
+  std::ofstream outfile{ path, std::ios::out };
+  throwExceptionIf<InvalidDataException>(!outfile,
+                                         fmt::format("Could not open file {} to write", path.string()));
+
+  outfile.write(content.data(), static_cast<std::streamsize>(content.size()));
+}
+
+void writeBufferToFile(const std::filesystem::path& path, std::span<const std::byte> content) {
+  std::ofstream outfile{ path, std::ios::out | std::ios::binary };
+  throwExceptionIf<InvalidDataException>(!outfile,
+                                         fmt::format("Could not open file {} to write", path.string()));
+
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+  outfile.write(reinterpret_cast<const char*>(content.data()), static_cast<std::streamsize>(content.size()));
+}
+
+// -------------------------------------------------------------------------------------------------
+// ScopedPath implementation
+// -------------------------------------------------------------------------------------------------
+
+ScopedPath::ScopedPath(std::filesystem::path path) : path_(std::move(path)), path_str_(path_.string()) {
+}
+
+ScopedPath::~ScopedPath() {
+  std::error_code e;
+  std::filesystem::remove_all(path_, e);
+  if (e) {
+    fmt::print(stderr, "failed to remove test temporary directory {}, with error: {}\n", path_.string(),
+               e.message());
+  }
+}
+
+auto ScopedPath::createFile() -> ScopedPath {
+  const auto global_temp_dir = std::filesystem::temp_directory_path();
+  auto file = global_temp_dir / randomString();
+  while (std::filesystem::exists(file)) {
+    file = global_temp_dir / randomString();
+  }
+
+  return ScopedPath{ file };
+}
+
+auto ScopedPath::createDir() -> ScopedPath {
+  const auto global_temp_dir = std::filesystem::temp_directory_path();
+  auto temp_dir = global_temp_dir / randomString();
+  while (std::filesystem::exists(temp_dir)) {
+    temp_dir = global_temp_dir / randomString();
+  }
+
+  std::filesystem::create_directory(temp_dir);
+  return ScopedPath{ std::move(temp_dir) };
```

Comment:
Not sure on the role of path. Do you want to use path here, or is it a top level path? 
Furthermore, indicate if you want a global or relative path if it matters (probably it does not).

Reply:
I am not sure I understand what you mean